File: core/src/main/java/org/elkd/core/Environment.java
package org.elkd.core;

import com.google.common.annotations.VisibleForTesting;
import org.elkd.shared.io.File;

/* Singleton */
public class Environment {
  private static final String ELKD_DEFAULT_HOME = "/usr/local/elkd";
  private static final String ELKD_HOME_VAR = "ELKD_HOME";

  private static Environment instance = null;

  public static Environment getInstance() {
    synchronized (Environment.class) {
      if (instance == null) {
        instance = new Environment();
      }
      return instance;
    }
  }

  private Environment() { }

  public String getHome() {
    final String value = getEnv(ELKD_HOME_VAR);
    return value != null
        ? value
        : ELKD_DEFAULT_HOME;
  }

  public String getHomeFilePath(final String fileName) {
    return File.join(getHome(), fileName);
  }

  public String getEnv(final String name) {
    return System.getenv(name);
  }
}

File: core/src/main/java/org/kerala/core/Environment.kt
package org.kerala.core

import org.kerala.core.config.Config
import org.kerala.core.config.ConfigProvider
import org.kerala.shared.io.File

/* Singleton */
object Environment {
  const val DEFAULT_HOME = "/usr/local/kerala"
  const val HOME_VAR = "KERALA_HOME"

  var args: Array<String> = emptyArray()
  val config: Config by lazy { ConfigProvider.config }

  private val home: String = this[HOME_VAR] ?: DEFAULT_HOME

  fun getHomePath(filePath: String): String {
    return File.join(home, filePath)
  }

  operator fun get(name: String): String? {
    return System.getenv(name)
  }
}
